ara_plumes/utils.py

```diff
@@ -15,13 +16,36 @@
 from .typing import Contour_List
 from .typing import Float2D
 from .typing import GrayImage
+from .typing import NpFlt
 from .typing import X_pos
 from .typing import Y_pos
 
 
-##################
-# Math Functions #
-##################
+# display utils
+def add_clock(f):
+    def _remove_underscore(name):
+        return name.replace("_", " ")
+
+    def _rename_f_sting(f_name):
+        if f_name == "video_to_ROM":
+            return "apply concentric circle"
+        if f_name.startswith("apply"):
+            return _remove_underscore(f_name)
+        return "apply " + _remove_underscore(f_name)
+
+    def clocked_f(*args, **kwargs):
+        f_name = _rename_f_sting(f.__name__)
+        print(f_name + ":", end="")
+        t0 = time.perf_counter()
+        result = f(*args, **kwargs)
+        elapsed_time = time.perf_counter() - t0
+        print("[%0.4fs]" % elapsed_time)
+        return result
+
+    return clocked_f
+
```

Comment:
This is a really cool decorator!  A few suggestions:
* `_remove_underscore(name)` is longer than `name.replace("_", " ")`, and the later is perfectly clear - this can be inlined
* If by `rename_f_sting` was a typo and you meant `rename_f_string`, I'd suggest a different name, since ["f-string"](https://docs.python.org/3/tutorial/inputoutput.html#formatted-string-literals) already has a specific meaning in python
* consider replacing `print()` with `logger.info()`